packages/app/src/cli/utilities/developer-platform-client/app-management-client.test.ts

```diff
@@ -187,3 +193,156 @@ describe('versionDeepLink', () => {
     expect(got).toEqual('https://dev.shopify.com/dashboard/1/apps/2/versions/3')
   })
 })
+
+describe('apiVersions', () => {
+  test('fetches available public API versions', async () => {
+    // Given
+    const mockedResponse: PublicApiVersionsQuery = {
+      publicApiVersions: [
+        {displayName: '2024-07', handle: '2024-07', supported: true},
+        {displayName: '2024-10', handle: '2024-10', supported: true},
+        {displayName: '2025-01 (Release candidate)', handle: '2025-01', supported: false},
+        {displayName: 'unstable', handle: 'unstable', supported: false},
+      ],
+    }
+    vi.mocked(webhooksRequest).mockResolvedValueOnce(mockedResponse)
+
+    // When
+    const client = new AppManagementClient()
+    client.token = () => Promise.resolve('token')
+    const apiVersions = await client.apiVersions()
+
+    // Then
+    expect(apiVersions.publicApiVersions.length).toEqual(mockedResponse.publicApiVersions.length)
+    expect(apiVersions.publicApiVersions).toEqual(mockedResponse.publicApiVersions.map((version) => version.handle))
+  })
+})
+
+describe('topics', () => {
+  test('fetches available topics for a valid API version', async () => {
+    // Given
+    const mockedResponse: AvailableTopicsQuery = {availableTopics: ['app/uninstalled', 'products/created']}
+    vi.mocked(webhooksRequest).mockResolvedValueOnce(mockedResponse)
+
+    // When
+    const client = new AppManagementClient()
+    client.token = () => Promise.resolve('token')
+    const topics = await client.topics({api_version: '2024-07'})
+
+    // Then
+    expect(topics.webhookTopics.length).toEqual(mockedResponse.availableTopics?.length)
+    expect(topics.webhookTopics).toEqual(mockedResponse.availableTopics)
+  })
+
+  test('returns an empty list when failing', async () => {
+    // Given
+    const mockedResponse: AvailableTopicsQuery = {availableTopics: null}
+    vi.mocked(webhooksRequest).mockResolvedValueOnce(mockedResponse)
+
+    // When
+    const client = new AppManagementClient()
+    client.token = () => Promise.resolve('token')
+    const topics = await client.topics({api_version: 'invalid'})
+
+    // Then
+    expect(topics.webhookTopics.length).toEqual(0)
+  })
+})
+
+describe('sendSampleWebhook', () => {
```

Comment:
I think maybe a thing relevant to test here is the mapping of `input` to the mutation variables